jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/validator/ResourceValidator.java

```diff
@@ -102,15 +105,17 @@ private void setupIgnoreRules(ResourceClassifier target) {
     public int validate() throws IOException {
         for(File resource: resources) {
             if (resource.isFile() && resource.exists()) {
-                try {
-                    log.info("validating %s resource", resource.toString());
-                    JsonNode inputSpecNode = geFileContent(resource);
-                    String kind = inputSpecNode.get("kind").toString();
-                    JsonSchema schema = getJsonSchema(prepareSchemaUrl(SCHEMA_JSON), kind);
-                    Set<ValidationMessage> errors = schema.validate(inputSpecNode);
-                    processErrors(errors, resource);
-                } catch (URISyntaxException e) {
-                    throw new IOException(e);
+                log.info("validating %s resource", resource.toString());
+                JsonNode inputSpecNode = geFileContent(resource);
+                String kind = inputSpecNode.get("kind").toString();
+                String[] allJars = ClassUtil.getJavaClassPath();
+                for (String jar : allJars) {
+                    URL schemaFile = ClassUtil.getFileFromJar(jar, SCHEMA_JSON);
+                    if (schemaFile != null) {
+                        JsonSchema schema = getJsonSchema(schemaFile, kind);
+                        Set<ValidationMessage> errors = schema.validate(inputSpecNode);
+                        processErrors(errors, resource);
+                    }
```

Comment:
Can be replaced by:

```suggestion
                for (URL schemaFile : Collections.list(ResourceValidator.class.getClassLoader().getResources(SCHEMA_JSON))) {
                    JsonSchema schema = getJsonSchema(schemaFile, kind);
                    Set<ValidationMessage> errors = schema.validate(inputSpecNode);
                    processErrors(errors, resource);
                }
```

No need for `ClassUtil` changes and manually checking Jar files (they should be available to the ClassLoader).